### ISSUE TITLE
2024-12-10-06

### ISSUE THREAD

**Issue by Arise-zwy** [Arise-zwy/Fed-Daily#1]
# Federated
## FIARSE: Model-Heterogeneous Federated Learning via Importance-Aware Submodel Extraction
- **Url**: http://arxiv.org/abs/2407.19389v3
- **Authors**: ['Feijie Wu', 'Xingchen Wang', 'Yaqing Wang', 'Tianci Liu', 'Lu Su', 'Jing Gao']
- **Abstrat**: In federated learning (FL), accommodating clients' varied computational capacities poses a challenge, often limiting the participation of those with constrained resources in global model training. To address this issue, the concept of model heterogeneity through submodel extraction has emerged, offering a tailored solution that aligns the model's complexity with each client's computational capacity. In this work, we propose Federated Importance-Aware Submodel Extraction (FIARSE), a novel approach that dynamically adjusts submodels based on the importance of model parameters, thereby overcoming the limitations of previous static and dynamic submodel extraction methods. Compared to existing works, the proposed method offers a theoretical foundation for the submodel extraction and eliminates the need for additional information beyond the model parameters themselves to determine parameter importance, significantly reducing the overhead on clients. Extensive experiments are conducted on various datasets to showcase the superior performance of the proposed FIARSE.

**Summary**: 

- (1):这篇文章的研究背景是在[联邦学习]（Federated Learning）中，客户端的计算能力不同，限制了部分资源受限客户端参与全局模型训练。

- (2):过去的方法包括静态和动态子模型提取，但存在问题是静态子模型无法适应模型参数重要性变化，动态子模型提取缺乏对参数重要性的指导。而提出的方法是根据模型参数的重要性动态调整子模型，避免了额外信息获取的开销，解决了现有方法的局限性，提出方法合理。

- (3):本文的贡献在于提出了一种基于重要性的模型异构联邦学习方法，能够有效构建适应客户端计算和存储能力的特定子模型，具备理论和实证上的优势。

- (4):文章提出的研究方法是基于重要性的子模型提取框架，通过模型参数的重要性水平构建客户端特定的子模型，避免了额外的存储或计算开销。

- (5):在图像和文本分类任务中，使用ResNet和RoBERTa模型，实验结果表明，提出的方法在性能上明显优于现有方法，特别是在资源受限设备上，支持了其目标。

## FedSynthCT-Brain: A Federated Learning Framework for Multi-Institutional Brain MRI-to-CT Synthesis
- **Url**: http://arxiv.org/abs/2412.06690v1
- **Authors**: ['Ciro Benito Raggio', 'Mathias Krohmer Zabaleta', 'Nils Skupien', 'Oliver Blanck', 'Francesco Cicone', 'Giuseppe Lucio Cascini', 'Paolo Zaffino', 'Lucia Migliorelli', 'Maria Francesca Spadea']
- **Abstrat**: The generation of Synthetic Computed Tomography (sCT) images has become a pivotal methodology in modern clinical practice, particularly in the context of Radiotherapy (RT) treatment planning. The use of sCT enables the calculation of doses, pushing towards Magnetic Resonance Imaging (MRI) guided radiotherapy treatments. Moreover, with the introduction of MRI-Positron Emission Tomography (PET) hybrid scanners, the derivation of sCT from MRI can improve the attenuation correction of PET images. Deep learning methods for MRI-to-sCT have shown promising results, but their reliance on single-centre training dataset limits generalisation capabilities to diverse clinical settings. Moreover, creating centralised multicentre datasets may pose privacy concerns.   To solve the issues, this study introduces FedSynthCT-Brain, a framework based on the Federated Learning (FL) paradigm for MRI-to-sCT in brain imaging.   We reproduced a federation across four European and American centres using a U-Net-based model. The approach was implemented using data from centres belonging the federation and it was tested on an unseen dataset from a centre outside the federation.   In the case of the unseen centre, the federated model achieved a median Mean Absolute Error (MAE) of 102.0 HU across 23 patients, with an interquartile range of 96.7-110.5 HU. The median (interquartile range) for the Structural Similarity Index (SSIM) and the Peak Signal to Noise Ratio (PNSR) were 0.89 (0.86-0.89) and 26.58 (25.52-27.42), respectively.   The analysis of the results showed acceptable performances of the federated approach, thus highlighting the potential of FL to enhance MRI-to-sCT to improve generalisability and advancing safe and equitable clinical applications while fostering collaboration and preserving data privacy.

**Summary**: 

- (1):本文研究背景包括合成计算机断层扫描图像在临床实践中的重要性，尤其在放射治疗规划中的作用；
 
- (2):过去的方法包括基于深度学习的MRI到sCT方法，但存在训练数据单一、泛化能力有限和隐私问题。本文提出的方法基于联邦学习范式，解决了这些问题，推动了MRI到sCT的发展；
 
- (3):本文的贡献在于提出了FedSynthCT-Brain，一个联邦学习框架用于多机构的脑MRI到CT的合成；
 
- (4):本文的研究方法是基于联邦学习的MRI到sCT框架，通过跨欧洲和美国四个中心的联合实现；
  
- (5):在未知中心的数据集上，该联邦模型实现了在23位患者中的中位绝对误差为102.0 HU，且SSIM中位值为0.89，PSNR中位值为26.58，表现可接受并支持其目标。

## Optimal estimation in private distributed functional data analysis
- **Url**: http://arxiv.org/abs/2412.06582v1
- **Authors**: ['Gengyu Xue', 'Zhenhua Lin', 'Yi Yu']
- **Abstrat**: We systematically investigate the preservation of differential privacy in functional data analysis, beginning with functional mean estimation and extending to varying coefficient model estimation. Our work introduces a distributed learning framework involving multiple servers, each responsible for collecting several sparsely observed functions. This hierarchical setup introduces a mixed notion of privacy. Within each function, user-level differential privacy is applied to $m$ discrete observations. At the server level, central differential privacy is deployed to account for the centralised nature of data collection. Across servers, only private information is exchanged, adhering to federated differential privacy constraints. To address this complex hierarchy, we employ minimax theory to reveal several fundamental phenomena: from sparse to dense functional data analysis, from user-level to central and federated differential privacy costs, and the intricate interplay between different regimes of functional data analysis and privacy preservation.   To the best of our knowledge, this is the first study to rigorously examine functional data estimation under multiple privacy constraints. Our theoretical findings are complemented by efficient private algorithms and extensive numerical evidence, providing a comprehensive exploration of this challenging problem.

**Summary**: 

- (1):这篇文章的研究背景是在功能数据分析中探讨差分隐私的保护。

- (2):过去的方法包括对功能均值进行估计和变系数模型估计，但存在隐私泄露和统计准确性之间的权衡问题。提出的方法引入了分布式学习框架和多层次隐私保护，通过最小化理论揭示不同隐私成本之间的权衡，解决了现有方法的问题，并具有充分的动机。

- (3):本文的贡献是在多重隐私约束下首次严格研究功能数据估计。

- (4):本文提出了基于最小二乘理论的复杂层次隐私保护框架，并结合高效私密算法和大量数值实验证据。

- (5):研究方法在功能数据估计任务中取得了良好性能，支持其研究目标。

## Faster Convergence on Heterogeneous Federated Edge Learning: An Adaptive Clustered Data Sharing Approach
- **Url**: http://arxiv.org/abs/2406.09776v3
- **Authors**: ['Gang Hu', 'Yinglei Teng', 'Nan Wang', 'Zhu Han']
- **Abstrat**: Federated Edge Learning (FEEL) emerges as a pioneering distributed machine learning paradigm for the 6G Hyper-Connectivity, harnessing data from the Internet of Things (IoT) devices while upholding data privacy. However, current FEEL algorithms struggle with non-independent and non-identically distributed (non-IID) data, leading to elevated communication costs and compromised model accuracy. To address these statistical imbalances within FEEL, we introduce a clustered data sharing framework, mitigating data heterogeneity by selectively sharing partial data from cluster heads to trusted associates through sidelink-aided multicasting. The collective communication pattern is integral to FEEL training, where both cluster formation and the efficiency of communication and computation impact training latency and accuracy simultaneously. To tackle the strictly coupled data sharing and resource optimization, we decompose the overall optimization problem into the clients clustering and effective data sharing subproblems. Specifically, a distribution-based adaptive clustering algorithm (DACA) is devised basing on three deductive cluster forming conditions, which ensures the maximum sharing yield. Meanwhile, we design a stochastic optimization based joint computed frequency and shared data volume optimization (JFVO) algorithm, determining the optimal resource allocation with an uncertain objective function. The experiments show that the proposed framework facilitates FEEL on non-IID datasets with faster convergence rate and higher model accuracy in a limited communication environment.

**Summary**: 

- (1):本文研究背景是在6G超连通性背景下的非独立同分布数据下的异构联合学习问题;
 
- (2):过去的方法包括模型驱动和数据驱动两种方法，存在通信成本高、模型准确性下降等问题。提出的方法通过聚类数据分享框架、自适应聚类算法和优化资源分配算法解决非IID数据下的统计不平衡问题，其动机合理;
 
- (3):本文的贡献是介绍了一种针对非IID数据的自适应聚类数据共享方法，提高了异构联合学习的收敛速度和模型准确性；
 
- (4):本文采用了分布式自适应聚类算法和基于随机优化的联合计算频率和共享数据量优化算法；
  
- (5):该方法在面对非IID数据时取得了更快的收敛速度和更高的模型准确性，支持其研究目标。

## A cautionary tale on the cost-effectiveness of collaborative AI in real-world medical applications
- **Url**: http://arxiv.org/abs/2412.06494v1
- **Authors**: ['Francesco Cremonesi', 'Lucia Innocenti', 'Sebastien Ourselin', 'Vicky Goh', 'Michela Antonelli', 'Marco Lorenzi']
- **Abstrat**: Background. Federated learning (FL) has gained wide popularity as a collaborative learning paradigm enabling collaborative AI in sensitive healthcare applications. Nevertheless, the practical implementation of FL presents technical and organizational challenges, as it generally requires complex communication infrastructures. In this context, consensus-based learning (CBL) may represent a promising collaborative learning alternative, thanks to the ability of combining local knowledge into a federated decision system, while potentially reducing deployment overhead. Methods. In this work we propose an extensive benchmark of the accuracy and cost-effectiveness of a panel of FL and CBL methods in a wide range of collaborative medical data analysis scenarios. The benchmark includes 7 different medical datasets, encompassing 3 machine learning tasks, 8 different data modalities, and multi-centric settings involving 3 to 23 clients. Findings. Our results reveal that CBL is a cost-effective alternative to FL. When compared across the panel of medical dataset in the considered benchmark, CBL methods provide equivalent accuracy to the one achieved by FL.Nonetheless, CBL significantly reduces training time and communication cost (resp. 15 fold and 60 fold decrease) (p < 0.05). Interpretation. This study opens a novel perspective on the deployment of collaborative AI in real-world applications, whereas the adoption of cost-effective methods is instrumental to achieve sustainability and democratisation of AI by alleviating the need for extensive computational resources.

**Summary**:

- (1):本文研究背景是针对在敏感医疗应用中协作人工智能的成本效益性进行警示性研究；

- (2):过去的方法包括联邦学习(FL)和基于共识的学习(CBL)，但实施FL存在技术和组织挑战，提出的方法是通过CBL减少训练时间和通信成本，解决了FL的部分问题，并且动机充分；

- (3):本文的贡献在于揭示了CBL作为FL的成本效益替代方案，提供了在医疗数据分析场景中等效精度的结果；

- (4):本文提出了对FL和CBL方法在多种医疗数据分析场景中的准确性和成本效益性进行广泛基准测试的研究方法；

- (5):本文在多种医疗数据分析任务中展示了CBL方法相比FL方法具有相当的准确性，并且显著减少了训练时间和通信成本（15倍和60倍减少），支持了他们的目标。

## Federated Split Learning with Model Pruning and Gradient Quantization in Wireless Networks
- **Url**: http://arxiv.org/abs/2412.06414v1
- **Authors**: ['Junhe Zhang', 'Wanli Ni', 'Dongyu Wang']
- **Abstrat**: As a paradigm of distributed machine learning, federated learning typically requires all edge devices to train a complete model locally. However, with the increasing scale of artificial intelligence models, the limited resources on edge devices often become a bottleneck for efficient fine-tuning. To address this challenge, federated split learning (FedSL) implements collaborative training across the edge devices and the server through model splitting. In this paper, we propose a lightweight FedSL scheme, that further alleviates the training burden on resource-constrained edge devices by pruning the client-side model dynamicly and using quantized gradient updates to reduce computation overhead. Additionally, we apply random dropout to the activation values at the split layer to reduce communication overhead. We conduct theoretical analysis to quantify the convergence performance of the proposed scheme. Finally, simulation results verify the effectiveness and advantages of the proposed lightweight FedSL in wireless network environments.

**Summary**:

- (1):这篇文章的研究背景是在无线网络中的联邦分裂学习（Federated Split Learning）；
 
- (2):过去的方法主要包括联邦学习和分裂学习，但存在模型规模增大导致边缘设备资源受限的问题。提出的方法通过对客户端模型进行修剪和梯度量化，以及随机丢弃中间激活值来减少计算和通信开销，从而解决了过去方法存在的问题，并且提出的方法具有很强的动机；
 
- (3):本文的贡献是提出了一种轻量级的联邦分裂学习方案，减轻了资源受限设备在调整大型模型时的存储和计算负担；
 
- (4):本文提出的研究方法是通过修剪模型参数、梯度量化和随机丢弃中间激活值的方式来减少资源受限设备上现有联邦分裂学习方法的计算和通信开销；
  
- (5):本文方法在无线网络环境中取得了良好的表现，加速了训练同时作为有效的正则化技术。

## SPEAR:Exact Gradient Inversion of Batches in Federated Learning
- **Url**: http://arxiv.org/abs/2403.03945v3
- **Authors**: ['Dimitar I. Dimitrov', 'Maximilian Baader', 'Mark Niklas Müller', 'Martin Vechev']
- **Abstrat**: Federated learning is a framework for collaborative machine learning where clients only share gradient updates and not their private data with a server. However, it was recently shown that gradient inversion attacks can reconstruct this data from the shared gradients. In the important honest-but-curious setting, existing attacks enable exact reconstruction only for batch size of $b=1$, with larger batches permitting only approximate reconstruction. In this work, we propose SPEAR, the first algorithm reconstructing whole batches with $b >1$ exactly. SPEAR combines insights into the explicit low-rank structure of gradients with a sampling-based algorithm. Crucially, we leverage ReLU-induced gradient sparsity to precisely filter out large numbers of incorrect samples, making a final reconstruction step tractable. We provide an efficient GPU implementation for fully connected networks and show that it recovers high-dimensional ImageNet inputs in batches of up to $b \lesssim 25$ exactly while scaling to large networks. Finally, we show theoretically that much larger batches can be reconstructed with high probability given exponential time.

**Summary**: 

- (1): 这篇文章研究的背景是[Federated Learning]中的梯度反演攻击问题。

- (2): 过去的方法仅能在小批量大小下实现梯度精确反演，而本文提出的方法SPEAR可以实现大于1的批量大小的梯度精确反演，通过利用梯度的低秩结构和ReLU激活函数的稀疏性来解决这一问题，提出的方法具有很好的动机。

- (3): 本文的贡献是首次提出一种可以在[Federated Learning]中实现大于1批量大小的梯度精确反演的算法SPEAR。

- (4): 本文提出的研究方法包括利用梯度的低秩结构和ReLU激活函数的稀疏性，以及采用采样法和贪婪选择方法进行梯度反演。

- (5): 本文的方法在高维度ImageNet输入下可以实现批量大小高达25的精确梯度反演，实验结果支持了他们的研究目标。

## H-FedSN: Personalized Sparse Networks for Efficient and Accurate Hierarchical Federated Learning for IoT Applications
- **Url**: http://arxiv.org/abs/2412.06210v1
- **Authors**: ['Jiechao Gao', 'Yuangang Li', 'Yue Zhao', 'Brad Campbell']
- **Abstrat**: The proliferation of Internet of Things (IoT) has increased interest in federated learning (FL) for privacy-preserving distributed data utilization. However, traditional two-tier FL architectures inadequately adapt to multi-tier IoT environments. While Hierarchical Federated Learning (HFL) improves practicality in multi-tier IoT environments by multi-layer aggregation, it still faces challenges in communication efficiency and accuracy due to high data transfer volumes, data heterogeneity, and imbalanced device distribution, struggling to meet the low-latency and high-accuracy model training requirements of practical IoT scenarios. To overcome these limitations, we propose H-FedSN, an innovative approach for practical IoT environments. H-FedSN introduces a binary mask mechanism with shared and personalized layers to reduce communication overhead by creating a sparse network while keeping original weights frozen. To address data heterogeneity and imbalanced device distribution, we integrate personalized layers for local data adaptation and apply Bayesian aggregation with cumulative Beta distribution updates at edge and cloud levels, effectively balancing contributions from diverse client groups. Evaluations on three real-world IoT datasets and MNIST under non-IID settings demonstrate that H-FedSN significantly reduces communication costs by 58 to 238 times compared to HierFAVG while achieving high accuracy, making it highly effective for practical IoT applications in hierarchical federated learning scenarios.

**Summary**:

- (1):本文研究背景为物联网(IoT)应用中的分层联邦学习(Hierarchical Federated Learning)存在的通信效率和准确性挑战。

- (2):过去的方法包括传统的两层FL架构和个性化联邦学习算法，但存在通信开销大、数据异质性、设备分布不均衡等问题。本文提出的方法(H-FedSN)通过引入二进制掩码机制和个性化层，减少通信开销，解决数据异质性和设备分布不均衡问题。该方法很有动机。

- (3):本文的贡献在于提出了针对实际IoT环境设计的H-FedSN方法，显著降低了通信成本并保持高准确性，适用于分层联邦学习场景。

- (4):本文提出的研究方法包括二进制掩码机制和个性化层的结构稀疏模型，以及贝叶斯聚合算法在边缘和云层级的应用。

- (5):本文在三个真实IoT数据集和MNIST数据集上进行实验，在非独立同分布的设置下，H-FedSN方法相对传统方法HierFAVG显著降低通信成本，同时保持高准确性，支持了他们的研究目标。
